File: Lab7 - Spring - Java/src/main/java/pl/edu/agh/to/school/student/Student.java
package pl.edu.agh.to.school.student;


import jakarta.persistence.*;
import pl.edu.agh.to.school.course.Course;
import pl.edu.agh.to.school.grade.Grade;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Entity
public class Student {
    @Id
    @GeneratedValue
    private int id;
    private String firstName;
    private String lastName;
    private LocalDate birthDate;
    private String indexNumber;
    @OneToMany
    private List<Grade> grades;

    @ManyToOne
    private Course course;


    public Student(int id, String firstName, String lastName, LocalDate birthDate, String indexNumber) {
        this.id = id;
        this.firstName = firstName;
        this.lastName = lastName;
        this.birthDate = birthDate;
        this.indexNumber = indexNumber;
        this.grades = new ArrayList<>();
    }

    public Student(String firstName, String lastName, LocalDate birthDate, String indexNumber) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.birthDate = birthDate;
        this.indexNumber = indexNumber;
        this.grades = new ArrayList<>();
    }

    public Student() {

    }

    public int getId() {
        return id;
    }

    public String getFirstName() {
        return firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public LocalDate getBirthDate() {
        return birthDate;
    }

    public String getIndexNumber() {
        return indexNumber;
    }

    public void giveGrade(Grade grade) {
        grades.add(grade);
    }

    public List<Grade> getGrades() {
        return grades;
    }

    public Course getCourse() {
        return course;
    }

    public void assignCourse(Course course) {
        this.course = course;
    }
}

File: Lab7 - Spring - Kotlin/src/main/kotlin/pl/edu/agh/to/school/student/Student.kt
package pl.edu.agh.to.school.student

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import pl.edu.agh.to.school.grade.Grade
import java.time.LocalDate

@Entity
class Student (
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private val id: Int,
    private val firstName: String,
    private val lastName: String,
    private val birthDate: LocalDate,
    private val indexNumber: String,
    private val grades: ArrayList<Grade>
    ){
    constructor() : this(0, "", "", LocalDate.now(), "", ArrayList())

    constructor(firstName: String, lastName: String, birthDate: LocalDate, indexNumber: String,grades: ArrayList<Grade>) : this(0, firstName, lastName, birthDate, indexNumber, grades)

    fun getId(): Int = id

    fun getFirstName(): String = firstName

    fun getLastName(): String = lastName

    fun getBirthDate(): LocalDate = birthDate

    fun getIndexNumber(): String = indexNumber

    fun getGrades(): List<Grade> = grades

    fun giveGrade(grade: Grade) {
        grades.add(grade)
    }

}